project/Dependencies.scala

```diff
@@ -68,6 +68,8 @@ object Dependencies {
       "com.typesafe.scala-logging" %% "scala-logging" % loggingVersion
     )
     val xgboostDep = "ml.dmlc" % "xgboost4j" % xgboostVersion // scala 2.11 only
+    val xgboostPredictorDep = "biz.k11i" % "xgboost-predictor" % "0.3.1"
```

Comment:
So the biz.k11i is the original version, you can check how many forks and stars it has compared to the h2oai. I am not sure why h2oai didn't 'github-fork' the project, I think they copied the code and init-ed a new github project, which is very confusing.

So, I've experimented with all of those:
- biz.k11i is the most widely used with 276 stars and 86 forks. Unfortunately it is not maintained anymore and it has a know problem (https://github.com/komiya-atsushi/xgboost-predictor-java/issues/21) because it uses doubles but should use floats (like xgboost4j does). @hollinwilkins was aware of this, so in this PR I'm forcibly casting to floats (like @hollinwilkins  was doing at the time). Check this comment that was left by @hollinwilkins, I dig it back from his PR and re-added it here future reference (https://github.com/combust/mleap/pull/645/files#r416048848)
- h2oai is a 'successor' of biz.k11i which is actively maintained and solves the float vs double bug. I'd love to use it but unfortunately they changed the API in some of the most recent ugprades, so it'd require some work to integrate. I can open an issue in MLEAP to have a look later on, if you'd like.
- On our side we use the Yelp's fork of biz.k11i (https://github.com/Yelp/xgboost-predictor-java) which fixes the double vs float issue and has an additional memory optimization. The problem in adopting the Yelp's biz.k11i version in MLeap is that we don't maintain it as much as h2oai does maintain their 'fork', so maybe their would be more forward-looking? Maybe we can have a look into that in the future..

but personally in the scope of this PR, I think the original biz.k11i is good enough

Reply:
agreed  biz.k11i is fine, good to have the context.